web/src/edit/NeighbourhoodMode.svelte

```diff
@@ -332,126 +330,157 @@
-      </button>
+        </button>
+        <button
+          on:click={() => (action = { kind: "main-roads" })}
+          disabled={action.kind == "main-roads"}
+          class="icon-btn"
+          class:active={action.kind == "main-roads"}
+          class:outline={action.kind != "main-roads"}
+          data-tooltip="Reclassify main roads (hotkey 5)"
+        >
+          <img src={mainRoadIconUrl} alt="Change main/minor roads" />
+        </button>
+      </div>
+      <div style="display: flex; justify-content: right; gap: 8px;">
+        <button
+          class="outline icon-btn"
+          disabled={undoLength == 0}
+          on:click={undo}
+          data-tooltip={undoLength == 0
+            ? "Undo Ctrl+Z"
+            : `Undo (${undoLength}) Ctrl+Z`}
+        >
+          <Undo />
+        </button>
+        <button
+          class="outline icon-btn"
+          disabled={redoLength == 0}
+          on:click={redo}
+          data-tooltip={redoLength == 0
+            ? "Redo Ctrl+Y"
+            : `Redo (${redoLength}) Ctrl+Y`}
+        >
+          <Redo />
+        </button>
+      </div>
     </div>
 
-    <button class="outline" on:click={() => (settingFilterType = true)}>
-      Change modal filter type
-    </button>
-    <ChangeFilterModal bind:show={settingFilterType} />
-
-    <div style="display: flex; justify-content: space-between;">
-      <button
-        class="outline"
-        disabled={undoLength == 0}
-        on:click={undo}
-        data-tooltip="Ctrl+Z"
-      >
-        {#if undoLength == 0}
-          Undo
-        {:else}
-          Undo ({undoLength})
-        {/if}
-      </button>
-      <button
-        class="outline"
-        disabled={redoLength == 0}
-        on:click={redo}
-        data-tooltip="Ctrl+Y"
-      >
-        {#if redoLength == 0}
-          Redo
-        {:else}
-          Redo ({redoLength})
-        {/if}
-      </button>
+    <div
+      style="min-height: 200px; padding-bottom: 16px; border-bottom: solid var(--pico-muted-border-color) 1px;"
+    >
+      {#if action.kind == "filter"}
+        <h3>Add modal filter</h3>
+        <p>
+          Modal filters restrict what kind of traffic can pass through a road
+          segment. Place them strategically to deter shortcuts through your
+          neighbourhood.
+        </p>
+        <ChangeFilterModal bind:show={settingFilterType} />
+        <div style="display: flex; gap: 8px; align-items: center;">
+          <button class="outline" on:click={() => (settingFilterType = true)}>
+            Change modal filter type
+          </button>
+
+          <button
+            on:click={action.freehand
+              ? () => (action = { kind: "filter", freehand: false })
+              : () => (action = { kind: "filter", freehand: true })}
+            class:active={action.freehand}
+            class:outline={!action.freehand}
+            data-tooltip="Add many modal filters along a line (hotkey 2)"
```

Comment:
```suggestion
            data-tooltip="Add many modal filters along a line"
```

The hotkey is gone now (which is a good choice)